orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfo.groovy

```diff
@@ -139,10 +137,10 @@ class PackageInfo {
       }
 
       if (packageName) {
-        // At least one matched package must be present it the context.
-        if (!matched) {
-          notMachedPrefix = prefix
-          notMatchedFileExtension = fileExtension
+
+        // If it hasn't been possible to match a package and allowMissingPackageInstallation is false raise an exception.
+        if (!matched && !request.get("allowMissingPackageInstallation")) {
```

Comment:
It's ok as is, but might be even better to build a full list of all package misses instead of throwing on the first miss. Otherwise, the user might be in a situation where they make a config change in response to a thrown exception, and then get a slightly different exception at the same spot on the next iteration.